### PR TITLE
Fix conflicting `ChainWorkerActor`s when running benchmarks

### PR DESCRIPTION
**Pull request by jvff**
## Motivation

The command `linera benchmark` was hanging because multiple `ChainWorkerActor`s were created to handle the same chain. This led to them attempting to acquire the chain guard, and blocking forever when it failed. The duplicate actors were created because the same process created more than one `WorkerState` instance.

## Proposal

Use the local node in the client instead of creating temporary `WorkerState` instances.

## Test Plan

After this PR, `linera benchmark` now executes correctly.

## Release Plan

Fixes a user visible bug, so at least a new patch version should be released.

## Links

- [reviewer checklist](https://github.com/linera-io/linera-protocol/blob/main/CONTRIBUTING.md#reviewer-checklist)

### CODE REVIEW

File: linera-service/src/linera/main.rs [hunk 1]
```diff
@@ -669,10 +669,9 @@ impl Runnable for Job {
 
                 for rpc_msg in &proposals {
                     if let RpcMessage::BlockProposal(proposal) = rpc_msg {
-                        let (executed_block, _) =
-                            WorkerState::new("staging".to_string(), None, storage.clone())
-                                .stage_block_execution(proposal.content.block.clone())
-                                .await?;
+                        let executed_block = context
```
Comment (ma2bd) [inline linera-service/src/linera/main.rs:672]: Can we make the worker state private now?

**Review by ma2bd — COMMENTED** [review]
LGTM but need to fix CI

**Comment by afck** [comment]
Thanks! `linera benchmark` works for me with this.

File: linera-service/src/linera/main.rs (hunk 1 above)
Comment (jvff) [reply linera-service/src/linera/main.rs:672]: I don't think we can make `WorkerState` private (or rather `pub(crate)`) yet. It still seems to be used (`linera-rpc`, `linera-sdk` for tests, and `linera-service` still has one usage in the client).

**Review by andresilva91 — APPROVED** [review]
Accepting modulo CI
This k8s test issue (failing and retrying) seems to have been fixed, so maybe rebasing to trigger a rerun of the test should fix it?

**Pull request merged**